src/storage/src/hummock/sstable/block.rs

```diff
@@ -214,22 +214,26 @@ impl Block {
             CompressionAlgorithm::Lz4 => {
                 let mut decoder = lz4::Decoder::new(compressed_data.reader())
                     .map_err(HummockError::decode_error)?;
-                let mut decoded = Vec::with_capacity(uncompressed_capacity);
+                let mut decoded = BytesMut::with_capacity(uncompressed_capacity);
+                unsafe { decoded.set_len(uncompressed_capacity) }
                 decoder
-                    .read_to_end(&mut decoded)
+                    .read_exact(&mut decoded)
                     .map_err(HummockError::decode_error)?;
-                debug_assert_eq!(decoded.capacity(), uncompressed_capacity);
-                Bytes::from(decoded)
+
+                assert_eq!(uncompressed_capacity, decoded.len());
+                decoded.freeze()
```

Comment:
I check the implement of `Bytes` and I find that it will not allocate and copy any extra memory when convert `Vec` to `Bytes`. It just record the length of content. cc @MrCroxx

Reply:
I revert it first. It looks same when len == cap

Reply:
And, Enhanced assert limitations